Tests/UnitTests/TestHelpers/XCTestExtensions.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  XCTestExtensions.swift
+//
+//  Created by Nacho Soto on 4/13/22.
+
+import XCTest
+
+// Similar to `XCTUnrap` but it allows an `async` closure.
+func XCTAsyncUnwrap<T>(
+    _ expression: @autoclosure () async throws -> T?,
+    _ message: @autoclosure () -> String = "",
+    file: StaticString = #filePath,
+    line: UInt = #line
+) async throws -> T {
+    let value = try await expression()
+
+    return try XCTUnwrap(value)
```

Comment:
I couldn't just name it `XCTUnwrap` as an overload because `Swift` was attempting to invoke the method recursively instead of resolving to the correct overload.